mdata/store_cassandra.go

```diff
@@ -77,9 +78,37 @@ type cassandraStore struct {
 	session     *gocql.Session
 	writeQueues []chan *ChunkWriteRequest
 	readQueue   chan *ChunkReadRequest
+	ttlTables   map[uint32]string
 }
 
-func NewCassandraStore(addrs, keyspace, consistency, CaPath, Username, Password, hostSelectionPolicy string, timeout, readers, writers, readqsize, writeqsize, retries, protoVer int, ssl, auth, hostVerification bool) (*cassandraStore, error) {
+func (c *cassandraStore) ensureTtlTables(ttls []uint32, keyspace string) error {
+	for _, ttl := range ttls {
+		// calculate table name and window
+		var window int = 1
+		for tmp := ttl / (24 * 60 * 60); tmp > 1; tmp = tmp / 2 {
+			window *= 2
+		}
+		table_name := fmt.Sprintf("metric_%d", window)
+
+		// table isn't know to exist, ensure it exists
+		err := c.session.Query(fmt.Sprintf(table_schema, keyspace, table_name, window)).Exec()
```

Comment:
so the proposed `compaction-window-factor` is an operator-tunable knob so that the operator can make a tradeoff between ~20sstables and "many more" sstables, if their queries are mostly for small ranges.  The assumption here is that it's inefficient or suboptimal to read small subsets of data out of a single large sstable.
But is this assumption really true?
I just spent some time trying to get a better understanding of the impact of such a setup.
The recommendation to have fewer larger sstables seems almost universal, though I still haven't gonne to the bottom of it.
In particular I found the "memtable" and "sstable" section of 
http://manuel.kiessling.net/2016/07/11/how-cassandras-inner-workings-relate-to-performance/ quite interesting. In particular: 

> As said, the structure of an SSTable is optimized for efficient reads from disk – entries in one SSTable are sorted by partition key, and an index of all partition keys is loaded into memory when an SSTable is openend. Looking up a row therefore is only one disk seek, with further sequential reads for retrieving the actual row data – thus, no expensive random I/O needs to be performed.

So the larger the sstable, the larger the index it has to load into memory (are these cached by cassandra?), and it can find the row for the given partition (metric key) very fast.
However since we use wide rows it still has to seek to the right ranges of columns that fall within the requested time range.  I'm not sure exactly how that works. we use `ts` as clustering key.

Anyway this needs some more looking into, but I would be cautious introducing a config option if none of us anticipates using it.